yt/frontends/amrvac/data_structures.py

```diff
@@ -373,10 +376,14 @@ def _set_code_unit_attributes(self):
 
         # note: yt sets hydrogen mass equal to proton mass, amrvac doesn't.
         mp_cgs = self.quan(1.672621898e-24, "g")  # This value is taken from AstroPy
-        He_abundance = 0.1  # hardcoded parameter in AMRVAC
 
         # get self.length_unit if overrides are supplied, otherwise use default
         length_unit = getattr(self, "length_unit", self.quan(1, "cm"))
+        if self._parfiles is not None:
+            namelist = read_amrvac_namelist(self._parfiles)
+            He_abundance = namelist.get("mhd_list", {}).get("he_abundance", 0.1)
+        else:
+            He_abundance = 0.1
```

Comment:
We can still inline it with my previous suggestion
```suggestion
     namelist = read_amrvac_namelist(self._parfiles)
     He_abundance = namelist.get("mhd_list", {}).get("he_abundance", 0.1)
```

Reply:
`read_amrvac_namelist` currently doesn't work on empty lists as you suggest, so we need a check either here or in `io.py`. Do you have a preference?